### PR TITLE
Automate rollout of Istio module

### PR DESCRIPTION
**Pull request by barchw**
**Description**

Changes proposed in this pull request:

- Create action to automate rollout of Istio module

**Related issues**

### CODE REVIEW

File: .github/workflows/rollout-module.yaml [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+      - name: Create module configuration
+        env:
+          CHANNEL: ${{ inputs.channel }}
+        working-directory: module-manifests/modules/istio/${{ inputs.channel }}
+        run: |
+          echo "Creating module configuration file:"
+          cat <<EOF | tee module-config.yaml
+          name: ${FULL_MODULE_NAME}
+          channel: ${CHANNEL}
+          version: ${TAG}
```
Comment (triffer) [inline .github/workflows/rollout-module.yaml:115]: For experimental channel we will need the version in the format of `${TAG}-${CHANNEL}` or hardcoded `${TAG}-experimental`

File: .github/workflows/rollout-module.yaml [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+          manifest: ${MANIFEST_FILENAME}
+          defaultCR: ${DEFAULT_CR_FILENAME}
+          annotations:
+            "operator.kyma-project.io/doc-url": "${{ vars.DOC_URL }}"
```
Comment (triffer) [inline .github/workflows/rollout-module.yaml:119]: Is there a reason why we don't have the doc-url hardcoded?

**Review by triffer — CHANGES_REQUESTED** [review]

File: .github/workflows/rollout-module.yaml (hunk 2 above)
Comment (barchw) [reply .github/workflows/rollout-module.yaml]: Yes, it might just be better to hardcode it

File: .github/workflows/rollout-module.yaml [hunk 3]
```diff
@@ -107,20 +110,36 @@ jobs:
...
-          channel: ${CHANNEL}
-          version: ${TAG}
-          manifest: ${MANIFEST_FILENAME}
-          defaultCR: ${DEFAULT_CR_FILENAME}
-          annotations:
-            "operator.kyma-project.io/doc-url": "${{ vars.DOC_URL }}"
-          moduleRepo: https://github.com/kyma-project/istio.git
-          moduleSecurityScanConfig: true
-          EOF
+          if "${CHANNEL}" == "experimental"; then
```
Comment (triffer) [inline .github/workflows/rollout-module.yaml:116]: I think we can simply avoid duplication and errors when this needs to be updated in the future by writing it like that:
```shell
if "${CHANNEL}" == "experimental"; then
  MODULE_VERSION="${TAG}-experimental"
else
  MODULE_VERSION="${TAG}"
fi

cat <<EOF | tee module-config.yaml
  name: ${FULL_MODULE_NAME}
  channel: ${CHANNEL}
  version: ${MODULE_VERSION}
  manifest: ${MANIFEST_FILENAME}
  defaultCR: ${DEFAULT_CR_FILENAME}
  annotations:
    "operator.kyma-project.io/doc-url": "${DOC_URL}"
  moduleRepo: https://github.com/kyma-project/istio.git
  moduleSecurityScanConfig: true
EOF
```

**Comment by barchw** [comment]
/wip
need to check labelling

**Review by triffer — APPROVED** [review]

**Pull request merged**